src/compiler/checker.ts

```diff
@@ -14609,7 +14609,7 @@ namespace ts {
 
         function checkTypeOfExpression(node: TypeOfExpression): Type {
             checkExpression(node.expression);
-            return stringType;
+            return getUnionType(arrayFromMap(typeofEQFacts.keys(), s => getLiteralTypeForText(TypeFlags.StringLiteral, s)));

```

Comment:
can this union type be precomputed and cached to avoid doing the same work multiple times?